src/cast_core/include/cast_core/actors/OptionsConversion.hpp

```diff
@@ -0,0 +1,243 @@
+// Copyright 2019-present MongoDB Inc.
```

Comment:
Nice on pulling this out

Reply:
Any changes besides location of the code?

Reply:
No this was just moving the code out. I'm not terrifically happy with the location but it will do for now.

I will cut a new ticket to refactor the other actors to use the conversion operators once thiis code is merged.